ompi/runtime/ompi_rte.c

```diff
@@ -86,6 +86,8 @@ static int _setup_proc_session_dir(char **sdir);
 static bool fns_init=false;
 static opal_tsd_tracked_key_t print_args_tsd_key;
 static char* opal_print_args_null = "NULL";
+static char* mysessiondir = NULL;
```

Comment:
Yes - because whoever wrote this code didn't fill in the opal_process_info structure correctly and skipped the topmost layer of the session directory. I don't want to change things that much as those values are used throughout the code base. All we need to know is that top layer solely so that we can clean it up during finalize. It isn't used anywhere else.